snippets/javascript/jscalculator.md

```diff
@@ -0,0 +1,48 @@
+# JS based simple calculator
+
+```js
+'use strict'
+
+let operator = '';
+let operation = '';
+let number1;
+let number2;
+
+// accept input from the user and check if it is a number 
+while(isNaN(number1)) {
+ number1 = parseFloat(prompt ('Enter the first number: '));
+}
+  operator = prompt('Enter valid operator ( either +, -, * or / ): ');  
+  while(!(operator === '+' || operator === '-' || operator === '*' || operator === '/')){
+    operator = prompt('Enter valid operator ( either +, -, * or / ): ');  
+  }
+  // accept input from the user and check if it is a number 
+  while(isNaN(number2)) {  
+    number2 = parseFloat(prompt ('Enter the second number: '));
+   }
+
+let result; // declaration of the variable.  
+  
+// use if, elseif and else keyword to define the calculator condition in JavaScript.  
+if (operator === '+') { // use + (addition) operator to add two numbers  
+    result = number1 + number2; 
+    operation = "The sum is : " ;
+}  
+else if (operator === '-') { // use -  (subtraction) operator to subtract two numbers  
+    result = number1 - number2;  
+    operation = "The difference is : " ;
+}  
+else if (operator === '*') { // use * (multiplication) operator to multiply two numbers  
+    result = number1 * number2;  
+    operation = "The product is : " ;
+}  
+else  {  
+  if (number2 !== 0){ //Check if the divider is zero or not 
+    result = number1 / number2; // use / (division) operator to divide two numbers 
+    operation = "The  quotient is : " ;
+  } else
+   alert('you cant divide by zero !!!') 
```

Comment:
you can add a `'` with `\'` in js

Reply:
When a user enters 0 as a second number, he/she will still see the alert on line 48. 
Can you think of a good way to prevent this?